addons/mrp/models/stock_move.py

```diff
@@ -69,6 +69,14 @@ class StockMove(models.Model):
     needs_lots = fields.Boolean('Tracking', compute='_compute_needs_lots')
     order_finished_lot_ids = fields.Many2many('stock.production.lot', compute='_compute_order_finished_lot_ids')
 
+    @api.model
+    def create(self, vals):
+        # quantity_done is only explicitly set when the user creates the move and specifies it manually
+        quantity_done = vals.get('quantity_done')
+        if vals.get('raw_material_production_id') and quantity_done is not None and quantity_done <= 0:
+            raise ValidationError(_('You have to consume positive quantities.'))
+        return super(StockMove, self).create(vals)
+
     @api.depends('active_move_line_ids.qty_done', 'active_move_line_ids.product_uom_id')
     def _compute_done_quantity(self):
         super(StockMove, self)._compute_done_quantity()
```

Comment:
Seems good but we are discussing with @sle-odoo and lap in order to know if we should use a constraint on stock move line >= 0 instead.